core/src/main/scala/kafka/utils/ToolsUtils.scala

```diff
@@ -79,4 +94,226 @@ object ToolsUtils {
     CommandLineUtils.printUsageAndExit(parser, message)
     throw new AssertionError("printUsageAndExit should not return, but it did.")
   }
+
+  def buildBootstrapMetadata(metadataVersion: MetadataVersion,
+                             metadataOptionalArguments: Option[ArrayBuffer[ApiMessageAndVersion]],
+                             source: String): BootstrapMetadata = {
+
+    val metadataRecords = new util.ArrayList[ApiMessageAndVersion]
+    metadataRecords.add(new ApiMessageAndVersion(new FeatureLevelRecord().
+      setName(MetadataVersion.FEATURE_NAME).
+      setFeatureLevel(metadataVersion.featureLevel()), 0.toShort));
+
+    metadataOptionalArguments.foreach { metadataArguments =>
+      for (record <- metadataArguments) metadataRecords.add(record)
+    }
+
+    BootstrapMetadata.fromRecords(metadataRecords, source)
+  }
+
+  def parseArguments(args: Array[String]): Namespace = {
```

Comment:
It looks like these methods are specific to the Kafka storage tool. We should not put them in a utils class and not in core.